### PR TITLE
Gradopaw/lesson 04

### PR DESCRIPTION
**Pull request by akpoik**
@gradopaw

### CODE REVIEW

**Comment by akpoik** [comment]
@gradopaw

File: students/kulon_aleksander/lesson_04/unit_tests_distance.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import math
```
Comment (gradopaw) [inline students/kulon_aleksander/lesson_04/unit_tests_distance.py]: Don't mix ```unitest``` and ```pytest``` modules. Rewrite this by using only the pytest module. Remember about the proper naming convention.

File: students/kulon_aleksander/lesson_04/unit_tests_distance.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import unittest
```
Comment (gradopaw) [inline students/kulon_aleksander/lesson_04/unit_tests_distance.py]: Line 2 to be deleted

File: students/kulon_aleksander/lesson_04/unit_tests_distance.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import pytest
+from length_of_the_segment import distance
+
+
+class TestDistanceFunction(unittest.TestCase):
```
Comment (gradopaw) [inline students/kulon_aleksander/lesson_04/unit_tests_distance.py]: class TestDistanceFunction(object):

File: students/kulon_aleksander/lesson_04/unit_tests_distance.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    def test_input_none(self):
+        with pytest.raises(TypeError):
+            distance(None)
```
Comment (gradopaw) [inline students/kulon_aleksander/lesson_04/unit_tests_distance.py]: I supposed that this test not working as you designed. You have to pass four parameters to ```distance``` function - you passed only one and that's the reason for the raised TypeError exception. Try to run ```distance(None, 1, 2, 3)```

File: students/kulon_aleksander/lesson_04/unit_tests_distance.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+    def test_input_string(self):
+        with pytest.raises(TypeError):
+            distance('aoeu')
```
Comment (gradopaw) [inline students/kulon_aleksander/lesson_04/unit_tests_distance.py]: As above

File: students/kulon_aleksander/lesson_04/unit_tests_distance.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+    def test_zero_distance(self):
+        self.assertEqual(distance(1, 1, 1, 1), 0, "zero distance")
```
Comment (gradopaw) [inline students/kulon_aleksander/lesson_04/unit_tests_distance.py]: Write this using the pytest module same as the rest of test from this module

File: students/kulon_aleksander/lesson_04/unit_tests_fibonacci.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
+from fibonacci_numbers import fib
```
Comment (gradopaw) [inline students/kulon_aleksander/lesson_04/unit_tests_fibonacci.py]: Same comments as for students/kulon_aleksander/lesson_04/unit_tests_distance.py

**Comment by akpoik** [comment]
@gradopaw

File: students/kulon_aleksander/lesson_04/test_length_of_the_segment.py [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    def test_input_string(self):
+        with pytest.raises(TypeError):
+            distance('aoeu', 1, 2, 3)
+
+    def test_zero_distance(self):
+        assert distance(1, 1, 1, 1) == 0
+
+    def test_corner_cases(self):
+        assert distance(-1, -1, -2, -2) == math.sqrt(2)
+        assert distance(0, 1, 0, 2) == 1
```
Comment (gradopaw) [inline students/kulon_aleksander/lesson_04/test_length_of_the_segment.py]: Tip: Don't make too many assertions per test. Better is to have more tests than one big unit test that is checking few things. A test should only be able to fail for a single reason, namely that the thing it tests doesn’t behave as expected.

**Pull request merged**